beacon-chain/sync/rpc_send_request.go

```diff
@@ -129,3 +134,65 @@ func SendBeaconBlocksByRootRequest(
 	}
 	return blocks, nil
 }
+
+func SendBlobSidecarByRoot(
+	ctx context.Context, tor blockchain.TemporalOracle, p2pApi p2p.P2P, pid peer.ID,
+	ctxMap ContextByteVersions, req *p2ptypes.BlobSidecarsByRootReq,
+) ([]*pb.BlobSidecar, error) {
+	topic, err := p2p.TopicFromMessage(p2p.BlobSidecarsByRootName, slots.ToEpoch(tor.CurrentSlot()))
+	if err != nil {
+		return nil, err
+	}
+	log.WithField("topic", topic).Debug("Sending blob sidecar request")
+	stream, err := p2pApi.Send(ctx, req, topic, pid)
+	if err != nil {
+		return nil, err
+	}
+	defer closeStream(stream, log)
+
+	return readChunkEncodedBlobs(stream, p2pApi.Encoding(), ctxMap)
+}
+
+var ErrBlobChunkedReadFailure = errors.New("failed to read stream of chunk-encoded blobs")
+var ErrBlobUnmarshal = errors.New("Could not unmarshal chunk-encoded blob")
+
+func readChunkEncodedBlobs(stream network.Stream, encoding encoder.NetworkEncoding, ctxMap ContextByteVersions) ([]*pb.BlobSidecar, error) {
+	decode := encoding.DecodeWithMaxLength
+	max := int(params.BeaconNetworkConfig().MaxRequestBlobsSidecars)
+	var (
+		code uint8
+		msg  string
+		err  error
+	)
+	sidecars := make([]*pb.BlobSidecar, 0)
+	for i := 0; i < max; i++ {
+		code, msg, err = ReadStatusCode(stream, encoding)
+		if err != nil {
+			break
+		}
+		if code != 0 {
+			return nil, errors.Wrap(ErrBlobChunkedReadFailure, msg)
+		}
+		ctxb, err := readContextFromStream(stream)
+		if err != nil {
+			return nil, errors.Wrap(err, "error reading chunk context bytes from stream")
+		}
+
+		v, found := ctxMap[bytesutil.ToBytes4(ctxb)]
+		if !found {
+			return nil, errors.Wrapf(ErrBlobUnmarshal, fmt.Sprintf("unrecognized fork digest %#x", ctxb))
+		}
+		if v != version.Deneb {
```

Comment:
this would break for any forks after deneb

Reply:
This was intentional, because this function returns a deneb-specific type. I thought it was premature to add an interface type for blobs, so it seemed natural to make this break in an obvious way for future forks, so that we'll know we need to update this code to handle multiple forks.